client/test/execution.test.js

```diff
@@ -491,22 +461,26 @@ describe('Execution', function() {
 
     describe('Grid View', function() {
       it('should show full results in a grid', async function () {
-        var [historyEl] = await historyTest(this.test)
-        await historyEl.waitUntilExists('.results tbody tr:nth-child(4)')
-
-        historyEl.textNodes('table tbody td:nth-child(1)').length.should.be.lessThan(12)
-        historyEl.textNodes('table thead th').slice(0,2).should.deep.equal(['ID', 'Type'])
-        await retry(() => historyEl.textNodes('table tbody td:nth-child(1)').should.deep.equal(
-          new Array(12).fill('').map((_, i) => String(i + 1))
-        ))
-        historyEl.textNodes('table tbody td:nth-child(2)').slice(0, 3).should.deep.equal([
-          'WorkflowExecutionStarted', 'DecisionTaskScheduled', 'DecisionTaskStarted'
-        ])
-        historyEl.textNodes('table tbody td:nth-child(3)').should.deep.equal([
-          moment(fixtures.history.emailRun1[0].timestamp).format('MMM Do h:mm:ss a'),
-          '', '', '1s (+1s)', '2s (+1s)', '3s (+1s)', '8s (+5s)', '19s (+11s)',
-          '30s (+11s)', '41s (+11s)', '52s (+11s)', '1m 4s (+12s)'
-        ])
+        historyTest(this.test)
+          .then(async ([historyEl]) => {
```

Comment:
Curious as to why the `await historyTest` didn't work. This code seems like it's exactly the same except you went back to using promises. Perhaps you should make historyTest work with await instead?

Reply:
Not sure exactly why. To be honest I don't like integration tests to do this much testing. Ideally would prefer to have many unit tests that test functionality and 1-2 integration tests to make sure the main happy/unhappy path is working. Thats just my preference anyway but feel free to disagree.